disnake/interactions/base.py

```diff
@@ -530,6 +532,18 @@ async def send_message(
         InteractionResponded
             This interaction has already been responded to before.
         """
+        if allowed_mentions is not None:
+            if state.allowed_mentions is not None:
+                allowed_mentions = state.allowed_mentions.merge(allowed_mentions).to_dict()
+            else:
+                allowed_mentions = allowed_mentions.to_dict()
```

Comment:
Mixing different types into the same variable is a fairly questionable practice, I'd appreciate it if you kept the number of possible linting errors to 0.